Code/Tools/ProjectManager/tests/GemCatalogTests.cpp

```diff
@@ -35,30 +43,530 @@ namespace O3DE::ProjectManager
-        GemModel::SetIsAdded(*gemModel, indexA, true);
+        GemModel::SetIsAdded(*m_gemModel, indexA, true);
 
         // expect b and c are now dependencies of an added gem but not themselves added
         // cmake will handle dependencies
         EXPECT_TRUE(GemModel::IsAddedDependency(indexB) && GemModel::IsAddedDependency(indexC));
-        EXPECT_TRUE(!GemModel::IsAdded(indexB) && !GemModel::IsAdded(indexC));
+        EXPECT_FALSE(GemModel::IsAdded(indexB) || GemModel::IsAdded(indexC));
 
-        QVector<QModelIndex> gemsToAdd = gemModel->GatherGemsToBeAdded();
+        QVector<QModelIndex> gemsToAdd = m_gemModel->GatherGemsToBeAdded();
         EXPECT_TRUE(gemsToAdd.size() == 1);
         EXPECT_EQ(GemModel::GetName(gemsToAdd.at(0)), gemA.m_name);
     }
+
+    class GemCatalogFilterTests
+        : public GemCatalogTests 
+    {
+    public:
+        void SetUp() override
+        {
+            GemCatalogTests::SetUp(); 
+            m_proxyModel.reset(new GemSortFilterProxyModel(m_gemModel.get()));
+        }
+
+        void TearDown() override
+        {
+            m_proxyModel.release();
+            GemCatalogTests::TearDown(); 
+        }
+
+    protected:
+        AZStd::unique_ptr<GemSortFilterProxyModel> m_proxyModel;
+    };
+
+    class GemCatalogSearchFilterTests
+        : public GemCatalogFilterTests
+    {
+    public:
+        void SetUp() override
+        {
+            GemCatalogFilterTests::SetUp();
+
+            GemInfo gemfilterName, gemfilterDisplayName, gemfilterCreator, gemfilterSummary, gemfilterFeature;
+
+            gemfilterName.m_name = "Name";
+            gemfilterDisplayName.m_name = "D";
+            gemfilterCreator.m_name = "C";
+            gemfilterSummary.m_name = "S";
+            gemfilterFeature.m_name = "F";
+
+            gemfilterDisplayName.m_displayName = "Display Name";
+            gemfilterCreator.m_creator = "Johnathon Doe";
+            gemfilterSummary.m_summary = "Unique Summary";
+            gemfilterFeature.m_features.append("Creative Feature");
+
+            m_gemRows.append(m_gemModel->AddGem(gemfilterName).row());
+            m_gemRows.append(m_gemModel->AddGem(gemfilterDisplayName).row());
+            m_gemRows.append(m_gemModel->AddGem(gemfilterCreator).row());
+            m_gemRows.append(m_gemModel->AddGem(gemfilterSummary).row());
+            m_gemRows.append(m_gemModel->AddGem(gemfilterFeature).row());
+        }
+
+    protected:
+        enum RowOrder
+        {
+            Name,
+            DisplayName,
+            Creator,
+            Summary,
+            Features
+        };
+
+        QVector<int> m_gemRows;
+    };
+
+    TEST_F(GemCatalogSearchFilterTests, GemCatalog_Filters_By_Search_String_Name_Only)
+    {
+        m_proxyModel->SetSearchString("Name");
+
+        EXPECT_TRUE(m_proxyModel->filterAcceptsRow(m_gemRows[Name], QModelIndex()));
+        EXPECT_TRUE(m_proxyModel->filterAcceptsRow(m_gemRows[DisplayName], QModelIndex()));
```

Comment:
Interesting that the "Name" only searches the "Name" and "DisplayName", but the "Display Name" only, just searches the "DisplayName" and not "Name"  - Is this intentional?

Reply:
Yes this is purposeful. Token matching is not supported and is specifically called about by a later test below called `GemCatalog_Filters_By_Search_String_No_Token_Support`

Not to say that it won't likely be supported in the future, it just isn't right now.